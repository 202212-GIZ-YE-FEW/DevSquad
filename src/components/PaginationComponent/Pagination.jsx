import classnames from "classnames";
import { useTranslation } from "next-i18next";
const Pagination = ({ items, pageSize, currentPage, onPageChange }) => {
    const { i18n } = useTranslation();
    const pagesCount = Math.ceil(items / pageSize); // 100/10

    if (pagesCount === 1) return null;
    const pages = Array.from({ length: pagesCount }, (_, i) => i + 1);

    return (
        <>
            <p className='py-3 font-Rubik  font-bold'> Pages</p>
            <ul
                className={classnames(
                    "sm:mx-32 inline-flex max-w-xs justify-center items-center border border-black h-8"
                )}
            >
                {/* Left navigation arrow */}

                <li
                    onClick={() => onPageChange(currentPage - 1)}
                    className={`${
                        currentPage > 1
                            ? "m-1 h-8"
                            : "bg-gray-200 opacity-50 cursor-not-allowed m-1 h-8 pointer-events-none"
                    }`}
                >
                    {i18n.language === "en" ? (
                        <svg
                            width='34'
                            height='34'
                            viewBox='0 0 34 34'
                            fill='none'
                            xmlns='http://www.w3.org/2000/svg'
                        >
                            <path
                                fill-rule='evenodd'
                                clip-rule='evenodd'
                                d='M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM9.75871 16.4101L19.9922 8.24435C20.1578 8.0879 20.3825 8 20.6167 8C21.1045 8 21.5 8.37352 21.5 8.83427V25.1657C21.5 25.387 21.4069 25.5992 21.2413 25.7556C20.8964 26.0815 20.3371 26.0815 19.9922 25.7556L9.75871 17.5899C9.41376 17.2641 9.41376 16.7359 9.75871 16.4101Z'
                                fill='#1A1A1A'
                                fill-opacity='0.7'
                            />
                        </svg>
                    ) : (
                        <svg
                            width='34'
                            height='34'
                            viewBox='0 0 34 34'
                            fill='none'
                            xmlns='http://www.w3.org/2000/svg'
                        >
                            <path
                                fill-rule='evenodd'
                                clip-rule='evenodd'
                                d='M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM13.3833 8C13.6175 8 13.8422 8.0879 14.0078 8.24435L24.2413 16.4101C24.5862 16.7359 24.5862 17.2641 24.2413 17.5899L14.0078 25.7556C13.6629 26.0815 13.1036 26.0815 12.7587 25.7556C12.5931 25.5992 12.5 25.387 12.5 25.1657V8.83427C12.5 8.37352 12.8955 8 13.3833 8Z'
                                fill='#1A1A1A'
                                fill-opacity='0.7'
                            />
                        </svg>
                    )}
                </li>

                {pages.map((page) => (
                    <li
                        key={page}
                        className={`flex items-center p-3 text-center h-8 min-w-6 border-2 border-black m-1 rounded-lg ${
                            page === currentPage
                                ? "bg-primary-orange text-white"
                                : "bg-white text-black"
                        }`}
                        onClick={() => onPageChange(page)}
                    >
                        <a>{page}</a>
                    </li>
                ))}
                {/*  Right Navigation arrow */}

                <li
                    onClick={() => onPageChange(currentPage + 1)}
                    className={`${
                        currentPage < pagesCount
                            ? "m-1 h-8"
                            : "bg-gray-200 opacity-50 cursor-not-allowed m-1 h-8 pointer-events-none"
                    }`}
                >
                    {i18n.language === "en" ? (
                        <svg
                            width='34'
                            height='34'
                            viewBox='0 0 34 34'
                            fill='none'
                            xmlns='http://www.w3.org/2000/svg'
                        >
                            <path
                                fill-rule='evenodd'
                                clip-rule='evenodd'
                                d='M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM13.3833 8C13.6175 8 13.8422 8.0879 14.0078 8.24435L24.2413 16.4101C24.5862 16.7359 24.5862 17.2641 24.2413 17.5899L14.0078 25.7556C13.6629 26.0815 13.1036 26.0815 12.7587 25.7556C12.5931 25.5992 12.5 25.387 12.5 25.1657V8.83427C12.5 8.37352 12.8955 8 13.3833 8Z'
                                fill='#1A1A1A'
                                fill-opacity='0.7'
                            />
                        </svg>
                    ) : (
                        <svg
                            width='34'
                            height='34'
                            viewBox='0 0 34 34'
                            fill='none'
                            xmlns='http://www.w3.org/2000/svg'
                        >
                            <path
                                fill-rule='evenodd'
                                clip-rule='evenodd'
                                d='M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM9.75871 16.4101L19.9922 8.24435C20.1578 8.0879 20.3825 8 20.6167 8C21.1045 8 21.5 8.37352 21.5 8.83427V25.1657C21.5 25.387 21.4069 25.5992 21.2413 25.7556C20.8964 26.0815 20.3371 26.0815 19.9922 25.7556L9.75871 17.5899C9.41376 17.2641 9.41376 16.7359 9.75871 16.4101Z'
                                fill='#1A1A1A'
                                fill-opacity='0.7'
                            />
                        </svg>
                    )}
                </li>
            </ul>
        </>
    );
};

export default Pagination;
