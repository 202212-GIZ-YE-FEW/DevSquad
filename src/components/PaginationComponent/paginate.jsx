export default function paginate(items, pageNumber, pageSize) {
    // get the pageNumber muins one and multiple by the page size 2 it will be 1-1 =0 *2 =0
    const startIndex = (pageNumber - 1) * pageSize;
    // get the events from the index 0 to the index 2 whitch will take to events and so on for the rest of events
    return items.slice(startIndex, startIndex + pageSize);
}
