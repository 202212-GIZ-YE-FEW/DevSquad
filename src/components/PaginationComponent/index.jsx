import React, { useMemo, useState } from "react";

import Pagination from "./Pagination";
import Eventcard from "../Eventcard";
// this number of recoded
let PageSize = 1;

const PaginationComponent = () => {
    const data = [
        {
            id: 999,
            eventImage: "/images/Rectangle2.png",
            eventDate: "FRI, JUL -7:00 PM GMT+3",
            eventTitle: "Title of the Event1",
            eventDetails:
                "Details about the event. Lorem ipsum dolor sit ametz consectetur adipiscing elit, sed do eiusmod tempor incididuntuyuuyii iyooyi Lorem ipsum dolor sit ame consectetur, adipisicing elit. Deleniti quos pariat nemo veritatis repudiandae error suscipit. Quas saepe vel cupiditate, ipsa adipisci excepturi animi magnam facere culpa aliquam asperiores!",
            eventAttendance: [],
        },
        {
            id: 1000,
            eventImage: "/images/Rectangle2.png",
            eventDate: "FRI, JUL -7:00 PM GMT+3",
            eventTitle: "Title of the Event2",
            eventDetails:
                "Details about the event. Lorem ipsum dolor sit ametz consectetur adipiscing elit, sed do eiusmod tempor incididuntuyuuyii iyooyi Lorem ipsum dolor sit ame consectetur, adipisicing elit. Deleniti quos pariat nemo veritatis repudiandae error suscipit. Quas saepe vel cupiditate, ipsa adipisci excepturi animi magnam facere culpa aliquam asperiores!",
            eventAttendance: [],
        },
    ];

    const onPageChange = (page) => {
        setCurrentPage(page);
    };
    const [currentPage, setCurrentPage] = useState(1);

    const currentTableData = useMemo(() => {
        const firstPageIndex = (currentPage - 1) * PageSize;
        const lastPageIndex = firstPageIndex + PageSize;
        return data.slice(firstPageIndex, lastPageIndex);
    }, [currentPage]);

    return (
        <>
            <div className='flex flex-col'>
                {/* this is example replase it with events list */}

                {currentTableData.map((item, index) => {
                    return (
                        <>
                            <div className='md:order-1 order-2'>
                                <Eventcard
                                    key={index}
                                    eventImage={item.eventImage}
                                    eventDate={item.eventDate}
                                    eventDetails={item.eventDetails}
                                    eventTitle={item.eventTitle}
                                    eventAttendance={item.eventAttendance}
                                />
                            </div>
                        </>
                    );
                })}
                <div className='md:order-2 order-1 text-center md:text-start'>
                    <Pagination
                        currentPage={currentPage}
                        totalCount={data.length}
                        pageSize={PageSize}
                        onPageChange={onPageChange}
                    />
                </div>
            </div>
        </>
    );
};

export default PaginationComponent;
