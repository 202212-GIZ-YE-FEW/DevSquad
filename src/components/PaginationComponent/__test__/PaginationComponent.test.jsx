import renderer from "react-test-renderer";

import PaginationComponent from "../Index";

it("renders correctly", () => {
    const tree = renderer.create(<PaginationComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});
