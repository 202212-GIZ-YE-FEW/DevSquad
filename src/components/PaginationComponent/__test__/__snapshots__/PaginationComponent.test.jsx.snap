// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-col"
>
  <div
    className="sm:order-1 order-2"
  >
    <div
      className="border-2 border-black rounded-lg md:px-4 mb-6 mt-3 font-Rubik "
    >
      <div
        className="flex md:justify-between md:flex-row flex-col justify-center items-center"
      >
        <div>
          <p
            className="pt-2"
          >
            FRI, JUL -7:00 PM GMT+3
          </p>
        </div>
        <div
          className=" pt-2"
        >
          <div
            className="flex flex-row items-center"
          >
            <div>
              <div
                class="relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
              <div
                class="sm:-left-4 -left-3 relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
              <div
                class="sm:-left-8 -left-6 relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
            </div>
            <p
              className="font-Rubik p-2"
            >
              +12 Attendance
            </p>
          </div>
        </div>
      </div>
      <div
        className="grid sm:grid-cols-3 align-items-center gap-4 p-4"
      >
        <div>
          <img
            alt="eventCardImage"
            className="rounded border-2 border-black w-64 md:h-36 h-24"
            data-nimg="1"
            decoding="async"
            height={380}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            responsive={true}
            src="/_next/image?url=%2Fimages%2FRectangle2.png&w=1920&q=75"
            srcSet="/_next/image?url=%2Fimages%2FRectangle2.png&w=750&q=75 1x, /_next/image?url=%2Fimages%2FRectangle2.png&w=1920&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={650}
          />
        </div>
        <div
          className="col-span-2 flex flex-col"
        >
          <p
            className="font-medium"
          >
            Title of the Event1
          </p>
          <p
            className="font-normal line-clamp-2"
          >
            Details about the event. Lorem ipsum dolor sit ametz consectetur adipiscing elit, sed do eiusmod tempor incididuntuyuuyii iyooyi Lorem ipsum dolor sit ame consectetur, adipisicing elit. Deleniti quos pariat nemo veritatis repudiandae error suscipit. Quas saepe vel cupiditate, ipsa adipisci excepturi animi magnam facere culpa aliquam asperiores!
          </p>
          <div
            className="flex justify-end py-5"
          >
            <button
              className="bg-primary-orange text-white rounded px-8 py-1 "
            >
              Join
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sm:order-1 order-2"
  >
    <div
      className="border-2 border-black rounded-lg md:px-4 mb-6 mt-3 font-Rubik "
    >
      <div
        className="flex md:justify-between md:flex-row flex-col justify-center items-center"
      >
        <div>
          <p
            className="pt-2"
          >
            FRI, JUL -7:00 PM GMT+3
          </p>
        </div>
        <div
          className=" pt-2"
        >
          <div
            className="flex flex-row items-center"
          >
            <div>
              <div
                class="relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
              <div
                class="sm:-left-4 -left-3 relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
              <div
                class="sm:-left-8 -left-6 relative inline-flex items-center justify-center sm:w-8 w-6 sm:h-8 h-6 bg-black rounded-full"
              >
                <span
                  class="text-white font-Rubik"
                >
                  R
                </span>
              </div>
            </div>
            <p
              className="font-Rubik p-2"
            >
              +12 Attendance
            </p>
          </div>
        </div>
      </div>
      <div
        className="grid sm:grid-cols-3 align-items-center gap-4 p-4"
      >
        <div>
          <img
            alt="eventCardImage"
            className="rounded border-2 border-black w-64 md:h-36 h-24"
            data-nimg="1"
            decoding="async"
            height={380}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            responsive={true}
            src="/_next/image?url=%2Fimages%2FRectangle2.png&w=1920&q=75"
            srcSet="/_next/image?url=%2Fimages%2FRectangle2.png&w=750&q=75 1x, /_next/image?url=%2Fimages%2FRectangle2.png&w=1920&q=75 2x"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={650}
          />
        </div>
        <div
          className="col-span-2 flex flex-col"
        >
          <p
            className="font-medium"
          >
            Title of the Event2
          </p>
          <p
            className="font-normal line-clamp-2"
          >
            Details about the event. Lorem ipsum dolor sit ametz consectetur adipiscing elit, sed do eiusmod tempor incididuntuyuuyii iyooyi Lorem ipsum dolor sit ame consectetur, adipisicing elit. Deleniti quos pariat nemo veritatis repudiandae error suscipit. Quas saepe vel cupiditate, ipsa adipisci excepturi animi magnam facere culpa aliquam asperiores!
          </p>
          <div
            className="flex justify-end py-5"
          >
            <button
              className="bg-primary-orange text-white rounded px-8 py-1 "
            >
              Join
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="md:order-2 order-1 text-center md:text-start"
  >
    <p
      className="py-3 font-Rubik  font-bold"
    >
       Pages
    </p>
    <ul
      className="sm:mx-32 inline-flex max-w-xs justify-center items-center border border-black h-8"
    >
      <button
        className="bg-gray-200 opacity-50 cursor-not-allowed m-1 h-8"
        disabled={true}
      >
        <svg
          fill="none"
          height="34"
          viewBox="0 0 34 34"
          width="34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM9.75871 16.4101L19.9922 8.24435C20.1578 8.0879 20.3825 8 20.6167 8C21.1045 8 21.5 8.37352 21.5 8.83427V25.1657C21.5 25.387 21.4069 25.5992 21.2413 25.7556C20.8964 26.0815 20.3371 26.0815 19.9922 25.7556L9.75871 17.5899C9.41376 17.2641 9.41376 16.7359 9.75871 16.4101Z"
            fill="#1A1A1A"
            fill-opacity="0.7"
            fill-rule="evenodd"
          />
        </svg>
      </button>
      <li
        className="flex items-center p-3 text-center h-8 min-w-6 border-2 border-black m-1 rounded-lg bg-primary-orange text-white"
        onClick={[Function]}
      >
        1
      </li>
      <li
        className="flex items-center p-3 text-center h-8 min-w-6 border-2 border-black m-1 rounded-lg bg-white text-black"
        onClick={[Function]}
      >
        2
      </li>
      <li
        className="m-1 h-8"
        onClick={[Function]}
      >
        <svg
          fill="none"
          height="34"
          viewBox="0 0 34 34"
          width="34"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M0.5 17C0.5 26.1127 7.8873 33.5 17 33.5C26.1127 33.5 33.5 26.1127 33.5 17C33.5 7.8873 26.1127 0.5 17 0.5C7.8873 0.5 0.5 7.8873 0.5 17ZM13.3833 8C13.6175 8 13.8422 8.0879 14.0078 8.24435L24.2413 16.4101C24.5862 16.7359 24.5862 17.2641 24.2413 17.5899L14.0078 25.7556C13.6629 26.0815 13.1036 26.0815 12.7587 25.7556C12.5931 25.5992 12.5 25.387 12.5 25.1657V8.83427C12.5 8.37352 12.8955 8 13.3833 8Z"
            fill="#1A1A1A"
            fill-opacity="0.7"
            fill-rule="evenodd"
          />
        </svg>
      </li>
    </ul>
  </div>
</div>
`;
